ts/Dashboards/Plugins/KPIComponent.ts

```diff
@@ -632,6 +634,28 @@ class KPIComponent extends Component {
         return '';
     }
 
+    public onDrop(sidebar: SidebarPopup, dropContext: Cell|Row): void|Cell {
+        if (sidebar && dropContext) {
+            const connectorsIds =
+                sidebar.editMode.board.dataPool.getConnectorIds();
+            let options: Partial<KPIComponent.ComponentOptions> = {
+                cell: '',
+                type: 'KPI'
+            };
+
+            if (connectorsIds.length) {
+                options = {
+                    ...options,
+                    connector: {
+                        id: connectorsIds[0]
+                    }
+                };
+            }
+
+            return sidebar.onDropNewComponent(dropContext, options);
+        }
+    }
```

Comment:
The body of the function is quite similar to other modules. I suggest creating a general function and making `options` as params. Then call it in the `onDrop` component.

Reply:
All the parameters here, among them `dropPointerSize`, should be optional. The default value is declared, the `|| 0`is only to trick the TS here 😉